interfaces/repo.go

```diff
@@ -537,17 +537,24 @@ func (r *Repository) connected(snapName, plugOrSlotName string) ([]ConnRef, erro
 	if r.plugs[snapName][plugOrSlotName] == nil && r.slots[snapName][plugOrSlotName] == nil {
 		return nil, fmt.Errorf("snap %q has no plug or slot named %q", snapName, plugOrSlotName)
 	}
-	// Collect all the relevant connections
+	// Collect all the relevant connections but be on the lookout for duplicates.
```

Comment:
Given the fixes we discussed and are planning on merging imminently, we should never have duplicates here, so this doesn't look right.